modules/pr-backend-deployment/variables.tf

```diff
@@ -0,0 +1,39 @@
+variable "repo" {
+  type        = string
+  description = "Full name of repo: organization/repository"
+}
+
+variable "cli_version" {
```

Comment:
it's good that you can provide version you want but I would set it to default as an empty string `""`.

In requirements you have:
> latest version v0.2.2 should be fetched from this [repo](https://github.com/selleo/cli)

so if cli_version is not provided by the user you should featch automatically the latest version using [this resource](https://registry.terraform.io/providers/integrations/github/latest/docs/data-sources/release)